partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelExclude.kt

```diff
@@ -0,0 +1,199 @@
+package org.partiql.eval.internal.operator.rel
+
+import org.partiql.eval.internal.Record
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.plan.Rel
+import org.partiql.plan.relOpExcludeTypeCollIndex
+import org.partiql.plan.relOpExcludeTypeCollWildcard
+import org.partiql.plan.relOpExcludeTypeStructKey
+import org.partiql.plan.relOpExcludeTypeStructSymbol
+import org.partiql.plan.relOpExcludeTypeStructWildcard
+import org.partiql.value.BagValue
+import org.partiql.value.CollectionValue
+import org.partiql.value.ListValue
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.SexpValue
+import org.partiql.value.StructValue
+import org.partiql.value.bagValue
+import org.partiql.value.listValue
+import org.partiql.value.sexpValue
+import org.partiql.value.structValue
+
+internal class RelExclude(
+    private val input: Operator.Relation,
+    private val exclusions: List<Rel.Op.Exclude.Path>
+) : Operator.Relation {
+
+    override fun open() {
+        input.open()
+    }
+
+    override fun next(): Record? {
+        val record = input.next() ?: return null
+        return exclusions.fold(record) { rec, path -> exclude(rec, path) }
+    }
+
+    override fun close() {
+        input.close()
+    }
+
+    @OptIn(PartiQLValueExperimental::class)
+    private fun exclude(
+        record: Record,
+        path: Rel.Op.Exclude.Path
+    ): Record {
+        val values = record.values
+        val value = values.getOrNull(path.root.ref)
+        val newValues = if (value != null) {
+            values[path.root.ref] = exclude(value, path.steps)
+            values
+        } else {
+            values
+        }
+        return Record(newValues)
```

Comment:
We shouldn't have missing roots and the record values array is mutable so we don't need to fold / combine previous states. The fold is useful when we need to produce a new immutable state, but we already have a mutable state object (record values) so we just need to apply each

```
    override fun next(): Record? {
        val record = input.next() ?: return null
        for (path in exclusions) {
             val i = path.root.ref
             val value = record.values[i]
             record.values[i] = exclude(value, path)
        }
        return record
    }
```

Reply:
Good catch. Removed the unnecessary fold